### PR TITLE
Add pydocstyle hook

### PR DESCRIPTION
**Pull request by AleCandido**
I wanted to start properly formatting docstrings, but I don't want to cause massive changes right now.

For this reason, I restricted the scope of `pydocstyle` to a minimal subset of checks. The plan is to gradually extend these checks to cover the full set.

To be merged after #482 (of course)

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.

### CODE REVIEW

**Comment by AleCandido** [comment]
> 1. why you prefer to start only considering public functions (`D103`)?

Public functions are the more relevant ones, since in principle are supposed to be used by the user. So, they **have to** be documented.

About internal functions: well, if they are documented so much the better, but at the moment I just wanted a minimal amount of warnings to fix ^^

**Comment by MatteoRobbiati** [comment]
> Public functions are the more relevant ones, since in principle are supposed to be used by the user. So, they **have to** be documented.

Ok, makes sense. What about e.g. `qibo.models.circuit.Circuit`? is it considered a public function or a class by `pydocstyle`? 

> About internal functions: well, if they are documented so much the better, but at the moment I just wanted a minimal amount of warnings to fix ^^

**Comment by AleCandido** [comment]
> 2. spaces instead of tabs is done for avoiding different tab's dimensions? Is it more ambiguous?

A good reason is also that is just the default, even for Black. Python code should be preferably indented with 4 spaces (not 2, 3, or 8).

If you want a discussion about the topic, you could read:
https://github.com/psf/black/issues/47

The main point is that a standard is needed. And *the standard* has to be only one. Someone chosen 4 spaces, and it's not that bad. Let's not create further mess :)

P.S.: I also prefer to use spaces, because I can move in columns inside the editor, while tabs are making a file a more irregular grid (the grid of characters does not correspond to the slots displayed, since one character could span multiple slots)
But even tabs have sense, everyone could choose what a tab corresponds to. So, maybe I prefer spaces just because I spent a lot of time programming in Python.

Another funny report of the debate:
https://www.alpharithms.com/tabs-vs-spaces-the-timeless-debate-581511/
mentioning an amusing SO blog post:
https://stackoverflow.blog/2017/06/15/developers-use-spaces-make-money-use-tabs/

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -32,3 +32,10 @@ repos:
     rev: 1.13.0
     hooks:
       - id: blacken-docs
+  - repo: https://github.com/pycqa/pydocstyle
+    rev: 6.3.0
+    hooks:
+      - id: pydocstyle
+        args:
+          - --select=D103,D200,D206,D300,D301
+        files: src/*
```
Comment (rodolfocarobene) [inline .pre-commit-config.yaml:41]:
```suggestion
        files: ^src/
```
Executing it locally I was getting a warning that is removed in this way

**Review by rodolfocarobene — APPROVED** [review]
There is just a small comment but, in general, nice!
Are you planning a big PR to solve all the doc issues and to extend the check or several smaller ones, adding few error codes every time?

**Comment by AleCandido** [comment]
> Are you planning a big PR to solve all the doc issues and to extend the check or several smaller ones, adding few error codes every time?

@rodolfocarobene the plan is to add a few keys every time. And when they will be enough, at some point we'll make a final batch (that I hope it will be as small as possible).

File: .pre-commit-config.yaml (hunk 1 above)
Comment (AleCandido) [reply .pre-commit-config.yaml]: Thanks :)

**Pull request merged**